pangeo_forge_esgf/utils.py

```diff
@@ -12,3 +14,24 @@ def facets_from_iid(iid: str, mip: str = None) -> Dict[str, str]:
     for name, value in zip(iid_name_template.split("."), iid.split(".")):
         facets[name] = value
     return facets
+
+
+def get_dataset_id_template(project: str, url: str = None):
+    """Requests the dataset_id string template for an ESGF project"""
+    if url is None:
+        url = "https://esgf-node.llnl.gov/esg-search/search"
+    params = {
+        "project": project,
+        "fields": "*",
+        "limit": 1,
+        "format": "application/solr+json"
+    }
+    r = requests.get(url, params)
+    #print(r.status_code)
+    return r.json()["response"]["docs"][0]["dataset_id_template_"][0]
+
+
+def facets_from_template(template: str):
+    """Parse the (dataset_id) string template into a list of (facet) keys"""
+    regex = r"\((.*?)\)"
+    return re.findall(regex, template)
```

Comment:
I dived a little bit into the [ESGF API documentation](https://esgf.github.io/esg-search/ESGF_Search_RESTful_API.html) and added now these two functions that should be able to retrieve the `dataset_id_template_` (and parse it into facets) for any ESGF project. I don't use them yet but i think this is better than what i did in the `params` module.